### DISCUSSION TITLE
How to use/port quartzNet model on mobile?

### DISCUSSION THREAD

**Discussion by rahul-hiker** [NVIDIA-NeMo/NeMo#1747] [Q&A] [answered]
Can I deploy NeMo trained ASR on mobile platform (Android/iOS)? If yes then what are the steps? Is there a working demo?

**Comment by okuchaiev** [comment, marked as answer]
> Can I deploy NeMo trained ASR on mobile platform (Android/iOS)? 
- Yes!

> If yes then what are the steps?
- Every NeMo model is inherited from ```torch.nn.Module``` , therefore you should checkout Pytorch tutorials about deploying models on mobile. For example, you can export QuartzNet ASR models into .onnx file for deployment.

> Is there a working demo?
- Unfortunately, we don't have an open-sourced working demo yet.

**Comment by rahul-hiker** [comment]
@okuchaiev Thanks for the reply.

I am looking for some answers in order to deploy the NeMo model on mobile.

1. How to export Nemo quartzNet model to .onnx?
2. When we deploy it on mobile then what are the libraries required (does it have some dependency on 3rd part libraries apart from pytorch code)?
3. What will be inference code in mobile?

**Comment by hoangtuanvu** [comment]
@rahul-hiker you can follow the instruction: https://pytorch.org/mobile/home/
I hope it helps you deploying Nemo model on mobile
